eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -179,11 +180,49 @@ case class Commitments(channelId: ByteVector32,
    * and our HTLC success in case of a force-close.
    */
   def almostTimedOutIncomingHtlcs(blockheight: Long, fulfillSafety: CltvExpiryDelta): Set[UpdateAddHtlc] = {
-    def nearlyExpired(add: UpdateAddHtlc) = blockheight >= (add.cltvExpiry - fulfillSafety).toLong
+    def nearlyExpired(add: UpdateAddHtlc): Boolean = blockheight >= (add.cltvExpiry - fulfillSafety).toLong
 
     localCommit.spec.htlcs.collect(incoming).filter(nearlyExpired)
   }
 
+  /** Compute our dust exposure in our commit tx (local) and their commit tx (remote). */
+  def currentDustExposure(): (MilliSatoshi, MilliSatoshi) = {
+    val localCommitDustExposure = CommitmentSpec.dustExposure(localCommit.spec, localParams.dustLimit, commitmentFormat)
+    val remoteCommitDustExposure = CommitmentSpec.dustExposure(remoteCommit.spec, remoteParams.dustLimit, commitmentFormat)
+    (localCommitDustExposure, remoteCommitDustExposure)
+  }
+
+  /** Test whether the given incoming htlc contributes to the dust exposure in the local or remote commit tx. */
+  def contributesToDustExposure(add: UpdateAddHtlc): (Boolean, Boolean) = {
+    val contributesToLocalCommitDustExposure = CommitmentSpec.contributesToDustExposure(OutgoingHtlc(add), localCommit.spec, localParams.dustLimit, commitmentFormat)
```

Comment:
Why is it an `OutgoingHtlc(add)` and not an incoming one ?

Reply:
Because incoming htlcs don't contribute to dust paid by us, since their cost is deducted from our peer's main output, not ours.
You can find more details on the spec PR here: https://github.com/lightningnetwork/lightning-rfc/pull/919